packages/account/src/Sections/Verification/ProofOfAddress/proof-of-address-container.tsx

```diff
@@ -107,59 +109,58 @@ const ProofOfAddressContainer = observer(() => {
     } = authentication_status;
 
     const from_platform = getPlatformRedirect(app_routing_history);
-
     const should_show_redirect_btn = Object.keys(platforms).includes(from_platform?.ref ?? '');
 
-    const redirect_button = should_show_redirect_btn && (
-        <Button
-            primary
-            className='proof-of-identity__redirect'
-            onClick={() => {
-                const url = platforms[from_platform.ref as keyof typeof platforms]?.url;
-                if (url) {
-                    window.location.href = url;
-                    window.sessionStorage.removeItem('config.platform');
-                }
-            }}
-        >
-            <Localize i18n_default_text='Back to {{platform_name}}' values={{ platform_name: from_platform.name }} />
-        </Button>
-    );
+    const is_submitted = has_submitted_poa && !poa_address_mismatch;
+
+    const status_content = getPOAStatusMessages(document_status, { needs_poi, is_submitted }, should_show_redirect_btn);
 
-    if (is_loading) return <Loading is_fullscreen={false} className='account__initial-loader' />;
-    if (
-        !allow_document_upload ||
-        (!is_age_verified && !allow_poa_resubmission && document_status === 'none' && is_mx_mlt)
-    )
-        return <NotRequired />;
-    if (has_submitted_poa && !poa_address_mismatch)
-        return <Submitted needs_poi={needs_poi} redirect_button={redirect_button} />;
-    if (
+    const is_resubmission_required =
         resubmit_poa ||
         allow_poa_resubmission ||
         (has_restricted_mt5_account &&
             document_status &&
-            ['expired', 'rejected', 'suspected'].includes(document_status)) ||
-        poa_address_mismatch
-    ) {
-        return <ProofOfAddressForm is_resubmit onSubmit={onSubmit} />;
+            [AUTH_STATUS_CODES.SUSPECTED, AUTH_STATUS_CODES.REJECTED, AUTH_STATUS_CODES.EXPIRED].some(
+                status => status === document_status
+            )) ||
+        poa_address_mismatch;
+
+    const should_show_poa_form =
+        is_age_verified &&
+        allow_document_upload &&
+        (document_status === AUTH_STATUS_CODES.NONE || is_resubmission_required);
+
+    const buttonOnclick = () => {
```

Comment:
```suggestion
    const handleClick = () => {
```
Will it be more appropriate to be a handler?

Reply:
thanks for suggestion,
I would say this, because it returns onclick for button, and not used for click :)